### ISSUE TITLE
[FEATURE REQUEST] Add support for SSE-C

### ISSUE THREAD

**Issue by mikefaraponov** [wal-g/wal-g#573]
https://docs.aws.amazon.com/AmazonS3/latest/dev/ServerSideEncryptionCustomerKeys.html

**Comment by theromis** [comment]
we need it too, any progress on it?

**Comment by theromis** [comment]
@moredure feature merged, request can be closed

**Comment by x4m** [comment]
No, not really. We also need to bump storages module and it's very desired that someone verify that it works.

**Comment by x4m** [comment]
Aaaand, we should update documentation.
